fixed issue #96

Riddhimaan-Senapati opened the pull request:
## Description
<!--- Provide a general summary of your changes. -->
<!--- Mention related issues, pull requests, or discussions with #<issue/PR/discussion ID>. -->
<!--- Tag people for whom this PR may be of interest using @<username>. -->
fixed issue #96 for customizabe failure messages by adding failure_message parameters to AutoEval,ResponseGenerator and CounterfactualGenerator classes


## Contributor License Agreement
<!--- Select all that apply by putting an x between the brackets: [x] -->
- [x] confirm you have signed the [LangFair CLA](https://forms.office.com/pages/responsepage.aspx?id=uGG7-v46dU65NKR_eCuM1xbiih2MIwxBuRvO0D_wqVFUMlFIVFdYVFozN1BJVjVBRUdMUUY5UU9QRS4u&route=shorturl)

## Tests
<!--- Select all that apply by putting an x between the brackets: [x] -->
- [x] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted

## Documentation
<!--- Select all that apply by putting an x between the brackets: [x] -->
- [x] no documentation changes needed
- [ ] README updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

## Screenshots
<!--- If applicable, please add screenshots. -->

Riddhimaan-Senapati commented:
@dylanbouchard @xavieryao @virenbajaj @dskarbrevik @vasisthasinghal Can anyone of you review this PR? I am excited to contribute to this repo.

dylanbouchard reviewed (commented):
Hi @Riddhimaan-Senapati - thanks for creating this PR! Have you tested this with a dictionary? It doesn't look like the code for the dictionary case is complete.

Riddhimaan-Senapati commented:
No, I haven't. Can you tell me what is wrong with the code?

dylanbouchard commented:
@virenbajaj can you take a look at your convenience?

Riddhimaan-Senapati commented:
@virenbajaj I have added support for failure_message being a dict in counterfactual.py and generator.py mainly by tweaking how non-completion rate is calculated in both files and handling what to return if an error occurs in _async_api_call() in generator.py

I didn't change much of auto.py to make it compatible for failure_message being a dict, except adding it as a parameter as I don't fully understand the code.

dylanbouchard commented:
> @virenbajaj @dylanbouchard I have added support for failure_message being a dict in counterfactual.py and generator.py mainly by tweaking how non-completion rate is calculated in both files and handling what to return if an error occurs in _async_api_call() in generator.py
> 
> I didn't change much of auto.py to make it compatible for failure_message being a dict, except adding it as a parameter as I don't fully understand the code.
> 
> Most of the code changes came from ruff, fixing the linting in other parts of the code.

Will take a look!

Riddhimaan-Senapati commented:
@dylanbouchard I made the changes that you required. Please review my code again. Thanks.

Riddhimaan-Senapati commented:
@dylanbouchard Have you reviewed my latest changes yet?

dylanbouchard commented:
> @dylanbouchard Have you reviewed my latest changes yet?

We will review later this week

Riddhimaan-Senapati commented:
@dylanbouchard Any updates yet?

dylanbouchard commented:
> @dylanbouchard Any updates yet?

@Riddhimaan-Senapati yes. @virenbajaj and I just reviewed. 

We decided that a more elegant design would be to have a dictionary as a third type that can be accepted for the `suppressed_exceptions` argument and to remove the `failure_message` parameter altogether. So your logic for when failure message is a dictionary would still work, but it would be specified with a different parameter instead.

Can you please refactor accordingly?

Riddhimaan-Senapati commented:
@dylanbouchard So what about when failure_message is a string, will we just be using FAILURE_MESSAGE constant instead?

dylanbouchard commented:
> @dylanbouchard So what about when failure_message is a string, will we just be using FAILURE_MESSAGE constant instead?

We can disregard that case. If they want to customize failure message(s), they can specify with a dict in `suppressed_exceptions`

Riddhimaan-Senapati commented:
@dylanbouchard so suppressed_exceptions can also be a dict like Dict[BaseException,str]. Is that correct?

dylanbouchard commented:
> @dylanbouchard so suppressed_exceptions can also be a dict like Dict[BaseException,str]. Is that correct?

exactly

Riddhimaan-Senapati commented:
@dylanbouchard Done, I have refactored the code to get rid of failure messages and just use suppressed_exceptions, I also added code to handle the case when suppressed_exceptions is just a tuple.

dylanbouchard commented:
> @dylanbouchard Done, I have refactored the code to get rid of failure messages and just use suppressed_exceptions, I also added code to handle the case when suppressed_exceptions is just a tuple.

Looks good. Running the unit tests now

Riddhimaan-Senapati commented:
@dylanbouchard I noticed that I hadn't changed the definitions of` suppressed_exceptions` in some places. Can you run the unit test again?

dylanbouchard commented:
@Riddhimaan-Senapati thank very much for your contribution! We should have a new release early next week, so you should show up as a contributor then.
